### PR TITLE
[ATMOSPHERE-263] feat: Support Kubernetes 1.29.x and 1.30.x

### PR DESCRIPTION
**Pull request by 0x00ace**
Successfully booted Kubernetes 1.27.11 and upgraded to new Kubernetes versions 1.27.11 -> 1.28.7 -> 1.29.2 using commands:
```
 openstack coe cluster upgrade cluster_1 k8s-v1.28.7
 openstack coe cluster upgrade cluster_1 k8s-v1.29.2
```

Also, tested new images uploading to the custom container registry without any issues

### CODE REVIEW

File: magnum_cluster_api/cmd/image_loader.py [hunk 1]
```diff
@@ -206,6 +209,7 @@ def _get_cloud_provider_images():
         "registry.k8s.io/provider-os/openstack-cloud-controller-manager:v1.28.0",
```
Comment (mnaser) [inline magnum_cluster_api/cmd/image_loader.py:209]: Could you please add the 1.28.2 images in here?

**Review by mnaser — CHANGES_REQUESTED** [review]

File: magnum_cluster_api/cmd/image_loader.py (hunk 1 above)
Comment (0x00ace) [reply magnum_cluster_api/cmd/image_loader.py:209]: Done - added full set of images

**Review by mnaser — CHANGES_REQUESTED** [review]
Sorry, last around of changes:

- We need to addthe 1.27.3 versions of images in `image-loader`

Once we have this, I will build and ship images for all these and switch the CI afterwards to include all the new images.

**Comment by 0x00ace** [comment]
@mnaser I have all sets of the images added. Also, I have updated k8s versions.

**Comment by mnaser** [comment]
@okozachenko1203 can you take into consideration your patch there?

**Comment by okozachenko1203** [comment]
@0x00ace thanks for your contribution.
i cannot make commits on this PR directly because it is in your personal fork. So i request to do changes yourself
could you import changes from this PR also? https://github.com/vexxhost/magnum-cluster-api/pull/353/files
- change the default `--version` flag value in `magnum_cluster_api/cmd/image_builder.py`
- update zuul CI jobs in `zuul.d` directory

**Comment by KeplerSysAdmin** [comment]
Kepler Team here,
We would like to have this k8s version supported ASAP.
Thank you

**Comment by KeplerSysAdmin** [comment]
v1.27 end of life is 2024-06-28
When can we merge this PR?

**Comment by mnaser** [comment]
recheck

**Comment by 0x00ace** [comment]
@mnaser 
I have successfully created a 1.28.x cluster and have upgraded it to 1.29.x
But to create a 1.29.x one needs to add a label keystone_auth_enabled=false

**Comment by okozachenko1203** [comment]
recheck

**Comment by 0x00ace** [comment]
For kubernetes >= 1.29.0 there is an [issue](https://github.com/kubernetes/cloud-provider-openstack/issues/2575) with k8s-keystone-auth

Because of that issue, I need to pass a label keystone_auth_enabled=false

**Comment by 0x00ace** [comment]
@mnaser 
Should I also include Kubernetes 1.30.x support in this PR? The issue with k8s-keystone-auth I have mentioned above is also valid for 1.30.x. Everything else looks to be working.

**Comment by cgroschupp** [comment]
It might be a good idea to split this PR to get a newer k8s version than 1.27 as soon as possible, since the end of life is in 6 days. And then work on the keystone-auth problems in another PR and release it later.

**Comment by mnaser** [comment]
I am struggling with finding a good way to manage the life cycle of these images other than "write a manual script" at this point.  It seems like it's hard to find a way to get CI to react for changes, and the image builds are flakey, and the Sonobuoy tests are flakey too, and on top of it they eat _a lot_ of resources...

**Comment by 0x00ace** [comment]
@mnaser I will rework this PR to support only k8s 1.29 and 1.30
k8s 1.28 support in this [PR](https://github.com/vexxhost/magnum-cluster-api/pull/403)

**Comment by mnaser** [comment]
@0x00ace thanks, actually I figured out a nice way to build images that is far more reliable.

https://github.com/vexxhost/capo-image-elements

I am almost done with the CI bit and then we can ignore the image build here and ship the images from there to consume here.

**Comment by mnaser** [comment]
Alright, 1.28 is in right now, I'm also trying to get image builds for EL9 working for 1.28, and then we can tackle this issue.

**Comment by 0x00ace** [comment]
@mnaser 
Updated PR for support k8s 1.29.x and 1.30.x
I'm using Rocky 9 with cilium and keystone_auth_enabled=false. 
Cluster creation, resizing, and adding new nodegroups look to be working without issues.
Also, I have no issues with a cluster upgrade from 1.28 -> 1.29 -> 1.30.

**Comment by mnaser** [comment]
https://github.com/vexxhost/capo-image-elements/issues/4

This is what I'm struggling for EL9

**Comment by okozachenko1203** [comment]
image
1.29 and 1.30 passed for calico at least

**Comment by okozachenko1203** [comment]
failed tests
```
1.29.6 cilium
[It] [sig-network] Services should serve endpoints on same port and different protocols [Conformance]

1.30 calico
[It] [sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should mutate configmap [Conformance]

1.30 cilium
[It] [sig-network] Services should serve endpoints on same port and different protocols [Conformance]
```

**Comment by okozachenko1203** [comment]
https://github.com/cilium/cilium/issues/29913

**Comment by okozachenko1203** [comment]
image
all ubuntu tests passed

**Comment by KeplerSysAdmin** [comment]
@okozachenko1203 
When is this version of magnum capi available?

**Pull request merged**